python/lsst/db/db.py

```diff
@@ -54,547 +41,115 @@
-    def tableExists(self, tableName, dbName=None):
-        """
-        Return True if table <tableName> exists in database <dbName>.
-
-        @param tableName  Table name.
-        @param dbName     Database name.
-
-        If <dbName> is not set, the current database name will be used.
-        """
-        dbNameStr = "'%s'" % dbName if dbName is not None else "DATABASE()"
-        cmd = "SELECT COUNT(*) FROM information_schema.tables "
-        cmd += "WHERE table_schema = %s AND table_name = '%s'" % \
-               (dbNameStr, tableName)
-        count = self.execCommand1(cmd)
-        return count[0] == 1
-
-    def createTable(self, tableName, tableSchema, dbName=None, mayExist=False):
-        """
-        Create table <tableName> in database <dbName>.
-
-        @param tableName   Table name.
-        @param tableSchema Table schema starting with opening bracket.
-        @param dbName      Database name.
-        @param mayExist    Flag indicating what to do if the database exists.
-
-        If database <dbName> is not set, and "use <database>" was called earlier,
-        it will use that database.
-        Raise exception if the table already exists and mayExist flag is not say to
-        True.
-        """
-        dbNameStr = "`%s`." % dbName if dbName is not None else ""
-        try:
-            self.execCommand0("CREATE TABLE %s`%s` %s" % \
-                              (dbNameStr, tableName, tableSchema))
-        except  DbException as e:
-            if e.errCode() == DbException.TB_EXISTS and mayExist:
-                self._log.debug("create table failed, mayExist is True")
-            else:
-                raise
-
-    def dropTable(self, tableName, dbName=None, mustExist=True):
-        """
-        Drop table <tableName> in database <dbName>.
-
-        @param tableName  Table name.
-        @param dbName     Database name.
-        @param mustExist  Flag indicating what to do if the database does not exist.
-
-        If <dbName> is not set, the current database name will be used. Raise
-        exception if the table does not exist and the mustExist flag is not set to
-        False.
-        """
-        dbNameStr = "`%s`." % dbName if dbName is not None else ""
-        try:
-            self.execCommand0("DROP TABLE %s`%s`" % (dbNameStr, tableName))
-        except DbException as e:
-            if e.errCode() == DbException.TB_DOES_NOT_EXIST and not mustExist:
-                self._log.debug("dropTable failed, mustExist is False")
+            raise DbException(DbException.UNSUPPORTED_DRIVER, driverName)
+        return self.getEngineFromArgs(**args)
+
+    def getEngineFromArgs(self, **kwArgs):
+        """
+        Initializes and returns SQLAlchemy engine using values provided through the
+        kwArgs. List of typically passed values:
+        - user or username
+        - passwd or password
+        - host
+        - port
+        - db or database
```

Comment:
Do we have to support all those "dual" options (user/username, passwd/password, db/database)? I'd probably just leave just one option and defined API without `**` but with all correct arguments (defaulted to None), it would be much easier to use/document.
One other thing - `create_engine` accepts a bunch of interesting keyword arguments (pool control, etc.), should we have an option to pass those options here?